stacker/blueprints/variables/types.py

```diff
@@ -1,15 +1,79 @@
 
 
+class TroposphereType(object):
+
+    def __init__(self, defined_type, resource_name=None):
+        """Represents a Troposphere type.
+
+        :class:`TroposphereType` will convert the value for the variable to the
+        given Troposphere type.
+
+        Args:
+            defined_type (Union[list, type]): List of or single Troposphere
+                type
+            resource_name (str): The name to use for the resource when creating
```

Comment:
Rather than this, what about making it so that every resource should have a `title` attribute that is required?  That's the name of the variable that is used in troposphere.  Another option is that we could do away with both the list & plain type syntax, and instead replace it with a single format that uses dictionaries like I did with the SecurityGroup blueprint: https://github.com/remind101/stacker_blueprints/pull/56/commits/ab094c32b1bcea2978796d6118bf43365571f2b7#diff-e1bf8fc9a0d5fa2e2b507057d3fcd3aaR29